### PR TITLE
Remote Cache Failure Logs #2081

### PR DESCRIPTION
**Pull request by claytoncasey01**
### Description

Output logs when remote cache fails

Fixes: #2081

### CODE REVIEW

**Comment by claytoncasey01** [comment]
@mehulkar Can you take a quick look at this? Ideally this wouldn't be passed down but I couldn't find another way to output the log to the console. This also might pose a problem for the all of the cache tests, none of those will run for me though even when pulling current main, they all fail.

File: cli/internal/cache/cache_http.go [hunk 1]
```diff
@@ -152,9 +175,13 @@ func (cache *httpCache) exists(hash string) (bool, int, error) {
...
 }
 
-func (cache *httpCache) retrieve(hash string) (bool, []turbopath.AnchoredSystemPath, int, error) {
+func (cache *httpCache) retrieve(
+	hash string,
+	prefixedUI *cli.PrefixedUi,
+) (bool, []turbopath.AnchoredSystemPath, int, error) {
 	resp, err := cache.client.FetchArtifact(hash)
 	if err != nil {
+		prefixedUI.Output(err.Error())
```
Comment (mehulkar) [inline cli/internal/cache/cache_http.go:184]: Is this the only material change? 

You'll want to do something like this:

```suggestion
		prefixedUI.Error(fmt.Sprintf("error fetching from remote cache: %s", err))
```

**Review by mehulkar — COMMENTED** [review]
I think this makes sense, but I'm not sure this code is used anymore or if all the caching got moved to Rust already. cc @NicholasLYang @chris-olszewski 

I'll note that the reason this is even necessary is because (at least in Go) errors from `cache_http.go Fetch()` get caught and don't bubbled up.

**Comment by claytoncasey01** [comment]
> I think this makes sense, but I'm not sure this code is used anymore or if all the caching got moved to Rust already. cc @NicholasLYang @chris-olszewski
> 
> I'll note that the reason this is even necessary is because (at least in Go) errors from `cache_http.go Fetch()` get caught and don't bubbled up.

The code seemed to still be hit when running locally against one of my projects that use turbo, all the errors I expected displayed.

**Comment by chris-olszewski** [comment]
> code is used anymore or if all the caching got moved to Rust

This is one of the codepaths that lives as both Go and Rust (when using `--experimental-rust-codepath`).

Due to the nature of this Go codepath going away soon and the fact that this will require changing the API for the Rust cache, I'm hesitant to encourage this until we're down to a single codepath. 

@claytoncasey01 If you can't wait and you've been looking to learn Rust, then I can provide some guidance on making this change on the Rust side.  The entrypoint to where you would want to start making changes would be [TaskCache::restore_outputs](https://github.com/vercel/turbo/blob/main/crates/turborepo-lib/src/run/cache.rs#L167)

**Comment by claytoncasey01** [comment]
> > code is used anymore or if all the caching got moved to Rust
> 
> This is one of the codepaths that lives as both Go and Rust (when using `--experimental-rust-codepath`).
> 
> Due to the nature of this Go codepath going away soon and the fact that this will require changing the API for the Rust cache, I'm hesitant to encourage this until we're down to a single codepath.
> 
> @claytoncasey01 If you can't wait and you've been looking to learn Rust, then I can provide some guidance on making this change on the Rust side. The entrypoint to where you would want to start making changes would be [TaskCache::restore_outputs](https://github.com/vercel/turbo/blob/main/crates/turborepo-lib/src/run/cache.rs#L167)

Thanks, I can take a look into making this on the Rust side, I'm pretty familiar with Rust and use it for a few different side projects. I'm also fine waiting, really just trying to familiarize myself with the Turbo codebase since I utilize turbo in my day job.

**Comment by mehulkar** [comment]
@claytoncasey01 implementing in Rust and Go in the same way would be ok, IMO

**Comment by anthonyshew** [comment]
Closing here since the repo is no longer in Go.
